src/sysusers/sysusers.c

```diff
@@ -163,6 +181,55 @@ static void maybe_emit_login_defs_warning(Context *c) {
         c->login_defs_need_warning = false;
 }
 
+static void log_audit_op(
+                Context *c,
+                int type,
+                const char *op,
+                const char *name,
+                unsigned id,
+                bool success) {
+#if HAVE_AUDIT
+        assert(c);
+
+        if (arg_dry_run || c->audit_fd < 0)
+                return;
+
+        log_trace("Emitting audit message: %s, %s, %u, %s",
+                  op, strna(name), id, success ? "good" : "bad");
+
+        audit_log_acct_message(
+                        c->audit_fd,
+                        type,
+                        program_invocation_short_name,
+                        op, name, id,
+                        /* host= */ NULL,
+                        /* addr= */ NULL,
+                        /* tty= */ NULL,
+                        success ? 1 : 0);
+#endif
+}
+
+static void log_audit_failures(
+                Context *c,
+                bool users_failed,
+                bool groups_failed) {
+        /* Generate audit events to describe the failure.
+         *
+         * The audit system expects individual failure messages for each user and group. We write the db one
+         * file at a time, so we can't say that one user or group "failed". Let's generate a failure message
+         * for each and every entry, matching the "add" events we emitted earlier.
+         */
+
+        Item *i;
+
+        if (users_failed)
+                ORDERED_HASHMAP_FOREACH(i, c->todo_uids)
+                        log_audit_op(c, AUDIT_ADD_USER, "adding user", i->name, -1, /* success= */ false);
```

Comment:
hmm, the 2nd to last param is "unsigned", yet you pass a negative value to it? also, should be prefixed `/* id=* /`

Reply:
th 3rd arg seems incorrectly formatted. quoting the man page:

```
              op  -  operation. Ex: "adding-user", "changing-finger-
              info", "deleting-group". This value should have a dash or
              underscore between the words so that report parsers group
              them together.
``

Reply:
Yeah. This is also what Steve Grubb mentioned in fedora-devel.